nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -159,31 +204,36 @@ def _setup_tokenizer(self, cfg: DictConfig):
         self.tokenizer = tokenizer
 
     def setup_training_data(self, train_data_config: Optional[DictConfig]):
-        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
+        self._train_dl = self._create_dataloader_from_config(cfg=train_data_config, mode='train')
 
     def setup_validation_data(self, val_data_config: Optional[DictConfig]):
-        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config)
+        self._validation_dl = self._create_dataloader_from_config(cfg=val_data_config, mode='validation')
 
     def setup_test_data(self, test_data_config: Optional[DictConfig]):
-        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config)
+        self._test_dl = self._create_dataloader_from_config(cfg=test_data_config, mode='test')
 
-    def _setup_dataloader_from_config(self, cfg: DictConfig):
-        input_file = cfg.file_name  # os.path.join(self.dataset_cfg.data_dir, f"{cfg.file_name}")
+    def _create_dataloader_from_config(self, cfg: DictConfig, mode):
```

Comment:
All models across collections use _setup_dataloader_from_config name, what is the reason for renaming?

Reply:
I reverted it back but I think it is not a good name as it says 'setup', but it creates and return the dataloaders while others methods with name "setup" actually set something in the model.

Reply:
I have experimented with a similar idea, also avoiding setup for the same reasons... so called it **instantiate_dataloader()**
https://github.com/tkornuta-nvidia/NeMo/blob/31ea5240fcef1c86de0307a7d87edcb03f11fd28/nemo/collections/cv/models/mnist_lenet5.py#L77